### ISSUE TITLE
Owls delivering quests inside houses

### ISSUE THREAD

**Issue by exectails** [aura-project/aura#225]
On officials, owls don't deliver quests inside houses, you need to go outside to receive them. We need to create a list in the creature's quest manager, with quests that are supposed to be sent by owl, and then check that list on region change.

Alternatively, we could add the quests, but with a flag, so they aren't displayed until region change. Otherwise we'd need a new db, to save the list across relogs.

**Comment by DeleteIT** [comment]
https://github.com/aura-project/aura/blob/master/src/ChannelServer/World/Quests/Quest.cs#L270

`public enum QuestState { Undelivered, InProgress, Complete }` ?

**Comment by exectails** [comment]
Hm, not a bad approach. Though we'd have to make sure all code still works, as something might've been written with the assumption that there are only two states. And Undelivered would have to go at the end, because the state is saved as int in the db, existing InProgress quests would become Undelivered.

Just realized, we also have to make sure not to finish objectives until it's delivered, and to do the initial item adds and objective checks afterwards.

**Comment by exectails** [comment]
While at it, we can also add a delay, so you can specify that a quest should be delivered after x minutes.

**Comment by ImaBrokeDude** [comment]
Good for generation questions too right?

**Comment by exectails** [comment]
I decided to go with my original idea, but with a slight change, as that made the most sense, after I had tried the alternatives. We now have a list of quest owls that are "waiting" to deliver quests to the player. They don't deliver until the player is outside and optionally until a delay is over.

On logout they are saved, on login loaded, and they are checked regularly, currently once every minute, to see if a delay is over. This way you'll always have a slight delay until owls deliver a quest, as it seems to be the case on official as well, and we can specific the hour long delays for the mainstream quests we all love so much^^
